src/client/Preview/Register.ts

```diff
@@ -3,47 +3,61 @@ import {TextDocumentContentProvider} from './TextDocumentContentProvider';
 
 export function registerPreview(context, window, client) {
 
+  const aureliaViewDataPanelType = 'aureliaViewData';
+
   let previewUri = vscode.Uri.parse('aurelia-preview://authority/aurelia-preview');
 
   let provider = new TextDocumentContentProvider(client);
   let registration = vscode.workspace.registerTextDocumentContentProvider('aurelia-preview', provider);
+  let isPanelVisible: boolean = false;
+  let panel: vscode.WebviewPanel;
+
+  function fillWebViewHtml(bodyContent: string): string {
+    return `
+      <!DOCTYPE html>
+      <html lang="en">
+        <head>
+            <meta charset="UTF-8">
+            <meta name="viewport" content="width=device-width, initial-scale=1.0">
+            <title>Cat Coding</title>
+        </head>
+        ${bodyContent}
+      </html>
+    `;
+  }
 
   vscode.workspace.onDidChangeTextDocument((e: vscode.TextDocumentChangeEvent) => {
+    if (!isPanelVisible) return;
     if (e.document === vscode.window.activeTextEditor.document) {
-      provider.update(previewUri);
+      provider.update(previewUri).then(success => {
+        panel.webview.html = fillWebViewHtml(success);
+      });
     }
   });
 
-  vscode.window.onDidChangeTextEditorSelection((e: vscode.TextEditorSelectionChangeEvent) => {
-    if (e.textEditor === vscode.window.activeTextEditor) {
-      provider.update(previewUri);
+  vscode.window.onDidChangeActiveTextEditor((editor: vscode.TextEditor) => {

```

Comment:
I was wondering why we need to update the webview after a selection changed, so instead, I changed it from
```diff
- onDidChangeTextEditorSelection
+ onDidChangeActiveTextEditor
```